cpp/tests/io/json_tree.cpp

```diff
@@ -837,7 +837,7 @@ TEST_P(JsonTreeTraversalTest, CPUvsGPUTraversal)
   }
   // gpu tree traversal
   auto [gpu_col_id, gpu_row_offsets] =
-    cuio_json::detail::records_orient_tree_traversal(d_input, gpu_tree, stream);
+    cuio_json::detail::records_orient_tree_traversal(d_input, gpu_tree, false, json_lines, stream);
```

Comment:
should we expand these tests as well?

Reply:
Not planned for this PR.

Reply:
is this because the orient does not impact the tree traversal?

Reply:
It's impacted. The additional test code is very similar to libcudf code.
I expanded the test, updated for array_of_arrays.